script.js

```diff
@@ -1,4 +1,165 @@
+//Global Variable
+var username =0
+var computerWin = 0
+var userWin = 0
+var roundCount = 0
+var gamemode = 0
+var firstWinner = 0     //0 = no lead, 1 = User lead, 2 = computer lead
+var ultiWinner = 0
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // input username
+  if (username == 0){
+    username = input
+    return "Hi "+username+" :) <br> Please select game mode  (normal/reversed/korean/computer)"
+  } else{   
+  // select game mode
```

Comment:
Great use of the boolean logics to change up your game modes. Displaying your understanding of the content very well here.